### PR TITLE
Port over kind's go_container.sh

### PR DESCRIPTION
**Pull request by amwat**
inspired from https://github.com/kubernetes-sigs/kind/blob/master/hack/

this will make local builds much more hermetic and updating go version very straightforward.

/cc @BenTheElder

### CODE REVIEW

**Comment by amwat** [comment]
https://github.com/kubernetes/test-infra/pull/18493 to add dind support for the e2e job.
plus we'll get to run the check on this PR itself.

File: hack/go_container.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# get the repo root for defaulting OUT_DIR and SOURCE_DIR
+REPO_ROOT="${REPO_ROOT:-$(git rev-parse --show-toplevel)}"
+# output directory, will be mounted to /out, defaults to /bin in REPO_ROOT
+OUT_DIR="${OUT_DIR:-${REPO_ROOT}/bin}"
+# source directory, will be mounted to /src, defaults to current directory
+SOURCE_DIR="${SOURCE_DIR:-$(pwd -P)}"
+# default to disabling CGO for easier reproducible builds and cross compilation
+export CGO_ENABLED="${CGO_ENABLED:-0}"
+# the container image, by default a recent official golang image
+GOIMAGE="${GOIMAGE:-golang:1.15beta1}"
```
Comment (BenTheElder) [inline hack/go_container.sh:33]: probably want rc1 now

File: hack/tools/go.sum [hunk 2]
```diff
@@ -0,0 +1,585 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (BenTheElder) [inline hack/tools/go.sum:1]: not sure if you did, but you'll want to `go mod tidy`,  it also minimizes the sum file a bit

File: hack/tools/tools.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+package tools
+
+
+import (
+	// linter(s)
+	_ "github.com/golangci/golangci-lint/cmd/golangci-lint"
+
+	// test runner
+	_ "gotest.tools/gotestsum"
+)
```
Comment (BenTheElder) [inline hack/tools/tools.go:16]: is this gofmt-ed?

File: hack/go_container.sh (hunk 1 above)
Comment (amwat) [reply hack/go_container.sh]: done

File: hack/tools/go.sum (hunk 2 above)
Comment (amwat) [reply hack/tools/go.sum:1]: done

File: hack/tools/tools.go (hunk 3 above)
Comment (amwat) [reply hack/tools/tools.go]: nice catch, seems lint wasn't running as expected. should be fixed now.

**Pull request merged**